File: src/server/src/main/java/com/project/application/models/exercise/CreateExerciseResource.java
package com.project.application.models.exercise;

import io.micronaut.serde.annotation.Serdeable;
import jakarta.annotation.Nonnull;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Serdeable
public record CreateExerciseResource(
        @Nonnull String name,
        @Nonnull String translatedName,
        @Nonnull String description,
        @Nonnull String gifUrl,
        @Nonnull String imageUrl,
        Optional<List<String>> mainMuscleGroupsIds,
        Optional<List<String>> synergisticMuscleGroupsIds,
        Optional<List<UUID>> variations,
        Optional<List<UUID>> equipment) {
}

File: src/server/src/main/java/com/project/application/models/exercise/CreateExerciseResource.kt
package com.project.application.models.exercise

import io.micronaut.serde.annotation.Serdeable
import jakarta.annotation.Nonnull
import java.util.Optional
import java.util.UUID

@Serdeable
data class CreateExerciseResource(@Nonnull val name: String,
                                  @Nonnull val translatedName: String,
                                  @Nonnull val description: String,
                                  @Nonnull val gifUrl: String,
                                  @Nonnull val imageUrl: String,
                                  val mainMuscleGroupsIds: Optional<List<String>>,
                                  val synergisticMuscleGroupsIds: Optional<List<String>>,
                                  val variations: Optional<List<UUID>>,
                                  val equipment: Optional<List<UUID>>)
